packages/endo/src/import.js

```diff
@@ -0,0 +1,58 @@
+/* global StaticModuleRecord */
+/* eslint no-shadow: 0 */
+
+import { compartmentMapForNodeModules } from "./compartmap.js";
+import { search } from "./search.js";
+import { assemble } from "./assemble.js";
+
+const decoder = new TextDecoder();
+
+const resolve = (rel, abs) => new URL(rel, abs).toString();
+
+const makeImportHookMaker = (read, rootLocation) => packageLocation => {
+  packageLocation = resolve(packageLocation, rootLocation);
+  return async moduleSpecifier => {
+    const moduleLocation = resolve(moduleSpecifier, packageLocation);
+    const moduleBytes = await read(moduleLocation);
+    const moduleSource = decoder.decode(moduleBytes);
+    return new StaticModuleRecord(moduleSource, moduleLocation);
+  };
+};
+
+export const loadPath = async (read, modulePath) => {
```

Comment:
There's a signature mismatch here.. if `importPath` is going to call this with extra unused arguments (maybe they're meant for future expansion?) then perhaps  `loadPath` should be defined to take `loadPath(read, modulePath, _endowments, _modules)`?